### PR TITLE
Pn 9310

### PR DESCRIPTION
**Pull request by azhansultan**
Refactored 9307 and 9310 feature files

### CODE REVIEW

File: src/main/java/it/pn/frontend/e2e/common/RecapitiDestinatarioPage.java [hunk 1]
```diff
@@ -65,19 +65,15 @@ public class RecapitiDestinatarioPage extends BasePage {
...
 
-    @FindBy(id = "legalContacts")
-    WebElement pecEmail;
-
     @FindBy(id = "pec")
     WebElement pecField;
 
+    @FindBy(id = "legalContacts")
+    WebElement pecEmail;
+
```
Comment (afus0) [inline src/main/java/it/pn/frontend/e2e/common/RecapitiDestinatarioPage.java:76]: there are some methods that take this id, please change them with this one that you add at the top.

File: src/main/java/it/pn/frontend/e2e/common/RecapitiDestinatarioPage.java [hunk 2]
```diff
@@ -351,12 +344,8 @@ public void clickSuEliminaPec() {
     public String waitLoadPopUpElimina() {
         By titlePopUp = By.id("dialog-title");
         By subTitlePopUp = By.id("dialog-description");
-        this.getWebDriverWait(10).withMessage("Non è stato caricato il titolo del modal").until(ExpectedConditions.visibilityOfElementLocated(titlePopUp));
-        this.getWebDriverWait(10).withMessage("Non è stato caricato il sottotitolo del modal").until(ExpectedConditions.visibilityOfElementLocated(subTitlePopUp));
-        this.getWebDriverWait(10).withMessage("Non è stato caricato il sottotitolo del modal").until(ExpectedConditions.visibilityOf(buttonAnnullaEliminazioneInPopUp));
-        this.getWebDriverWait(10).withMessage("Non è stato caricato il titolo del modal").until(ExpectedConditions.visibilityOf(buttonConfermaEliminazioneInPopUp));
```
Comment (afus0) [inline src/main/java/it/pn/frontend/e2e/common/RecapitiDestinatarioPage.java:357]: Why those have been removed?

File: src/main/java/it/pn/frontend/e2e/common/RecapitiDestinatarioPage.java [hunk 3]
```diff
@@ -365,18 +354,29 @@ public void clickSuConfermaElimina() {
...
     }
+    public boolean siControllaEliminazionePEC() {
+        try {
+            this.getWebDriverWait(10).until(ExpectedConditions.visibilityOf(pecField));
+            logger.info("pec non presente");
+            return true;
+        } catch (TimeoutException e) {
+            logger.error("pec presente con errore" + e.getMessage());
+            return false;
+        }
```
Comment (afus0) [inline src/main/java/it/pn/frontend/e2e/common/RecapitiDestinatarioPage.java]: I think you can minimize the code just by checking if it is displayed or not
```suggestion
        return pecField.isDisplayed();
```

File: src/main/java/it/pn/frontend/e2e/common/RecapitiDestinatarioPage.java [hunk 4]
```diff
@@ -365,18 +354,29 @@ public void clickSuConfermaElimina() {
...
         try {
             this.getWebDriverWait(10).until(ExpectedConditions.visibilityOf(pecEmail));
-            logger.info("pec presente");
+            logger.info("pec non presente");
             return true;
         } catch (TimeoutException e) {
-            logger.error("pec non presente con errore" + e.getMessage());
+            logger.error("pec presente con errore" + e.getMessage());
             return false;
         }
```
Comment (afus0) [inline src/main/java/it/pn/frontend/e2e/common/RecapitiDestinatarioPage.java]: same for this:
```java
public boolean siControllaPresenzaPEC() {
        return pecEmail.isDisplayed();
    }
```
Check if everything work as expected

File: src/main/java/it/pn/frontend/e2e/common/RecapitiDestinatarioPage.java [hunk 5]
```diff
@@ -560,7 +560,6 @@ public void checkButtonAnnullaEliminazioneInPopUp() {
     }
 
     public void clickButtonAnnullaEliminazioneInPopUp() {
-        this.getWebDriverWait(10).withMessage("Non è stato possibile cliccare sul bottone annulla").until(ExpectedConditions.elementToBeClickable(buttonAnnullaEliminazioneInPopUp));
```
Comment (afus0) [inline src/main/java/it/pn/frontend/e2e/common/RecapitiDestinatarioPage.java]: Why did you remove this condition?

File: src/test/java/it/pn/frontend/e2e/stepDefinitions/destinatario/personaFisica/RecapitiPersonaFisicaTest.java [hunk 6]
```diff
@@ -518,6 +518,22 @@ public void nellaPaginaITuoiRecapitiSiCliccaSulBottoneElimina() {
...
+        iTuoiRecapitiPage.eliminaEmailEsistente();
+        if (recapitiDestinatarioPage.waitLoadPopUpElimina().equalsIgnoreCase("Rimuovi e-mail")) {
+            recapitiDestinatarioPage.clickButtonAnnullaEliminazioneInPopUp();
+        } else {
+            recapitiDestinatarioPage.clickSuChiudiPopUp();
+            recapitiDestinatarioPage.eliminaNuovaPec();
+            recapitiDestinatarioPage.waitLoadPopUpElimina();
+            recapitiDestinatarioPage.clickButtonAnnullaEliminazioneInPopUp();
+        }
+    }
```
Comment (afus0) [inline src/test/java/it/pn/frontend/e2e/stepDefinitions/destinatario/personaFisica/RecapitiPersonaFisicaTest.java:535]: I think there are two separate steps that accomplish exactly what you're trying to do. Please verify.

File: src/test/java/it/pn/frontend/e2e/stepDefinitions/destinatario/personaFisica/RecapitiPersonaFisicaTest.java [hunk 7]
```diff
@@ -527,6 +543,17 @@ public void nellaPaginaITuoiRecapitiSiControllaCheLIndirizzoEmailNonSiaPresente(
...
+    public void nellaPaginaITuoiRecapitiSiControllaCheLIndirizzoEmailNonStataEleminata() {
+        logger.info("Si controlla che l'indirizzo Email non stata eleminata");
+        if (recapitiDestinatarioPage.verificaMailAssociata()) {
+            logger.info("L'email non è stata eliminata");
+        } else {
+            logger.error("L'email è stata eliminata");
+            Assert.fail("L'email è stata eliminata");
+        }
+    }
+
```
Comment (afus0) [inline src/test/java/it/pn/frontend/e2e/stepDefinitions/destinatario/personaFisica/RecapitiPersonaFisicaTest.java]: It seems similar to this: https://github.com/pagopa/pn-frontend-e2e/blob/4d0e2e2cc9224aa0d568f0da22892e280d12625d/src/test/java/it/pn/frontend/e2e/stepDefinitions/destinatario/personaFisica/RecapitiPersonaFisicaTest.java#L537-L544. The behavior you're trying to check is almost the same. See if you can find a workaround.

**Review by afus0 — CHANGES_REQUESTED** [review]

File: src/test/java/it/pn/frontend/e2e/stepDefinitions/destinatario/personaFisica/RecapitiPersonaFisicaTest.java (hunk 6 above)
Comment (azhansultan) [reply src/test/java/it/pn/frontend/e2e/stepDefinitions/destinatario/personaFisica/RecapitiPersonaFisicaTest.java:535]: it would be better to keep this steps together, as after check it could happen that popup is not "Rimuovi email", so in this case easier to manage popup, close it and reopen again

File: src/test/java/it/pn/frontend/e2e/stepDefinitions/destinatario/personaFisica/RecapitiPersonaFisicaTest.java (hunk 7 above)
Comment (azhansultan) [reply src/test/java/it/pn/frontend/e2e/stepDefinitions/destinatario/personaFisica/RecapitiPersonaFisicaTest.java]: yes. one expects verificaMailAssociata to be true, the other to be false. One controls that email is present, the other that it is deleted. it is two different steps that uses one common method, I didnt get what exactly to change?

File: src/test/java/it/pn/frontend/e2e/stepDefinitions/destinatario/personaFisica/RecapitiPersonaFisicaTest.java [hunk 8]
```diff
@@ -518,12 +518,41 @@ public void nellaPaginaITuoiRecapitiSiCliccaSulBottoneElimina() {
...
-            logger.error("Email non è stata eliminata correttamente");
-            Assert.fail("Email non è stata eliminata correttamente");
+            logger.info("Email è stata eliminata correttamente");
+        }else {
+            logger.error("Email non è stata eliminata");
+            Assert.fail("Email non è stata eliminata");
+        }
+    }
+
+    @And("Nella pagina I Tuoi Recapiti si controlla che l'indirizzo Email non stata eleminata")
```
Comment (afus0) [inline src/test/java/it/pn/frontend/e2e/stepDefinitions/destinatario/personaFisica/RecapitiPersonaFisicaTest.java]:
```suggestion
    @And("Nella pagina I Tuoi Recapiti si controlla che l'indirizzo Email non è stata eleminata")
```

**Review by afus0 — CHANGES_REQUESTED** [review]

File: src/test/java/it/pn/frontend/e2e/stepDefinitions/destinatario/personaFisica/RecapitiPersonaFisicaTest.java (hunk 6 above)
Comment (afus0) [reply src/test/java/it/pn/frontend/e2e/stepDefinitions/destinatario/personaFisica/RecapitiPersonaFisicaTest.java:535]: I was thinking of a proper way to manage both if it doesn't take too much time

**Review by afus0 — APPROVED** [review]
